apps/condo/domains/marketplace/schema/Invoice.test.js

```diff
@@ -591,6 +595,210 @@ describe('Invoice', () => {
+            const [resident] = await registerResidentByTestClient(
+                residentClient,
+                {
+                    address: client.property.address,
+                    addressMeta: client.property.addressMeta,
+                    unitType,
+                    unitName,
+                })
+
+            const [invoiceContext] = await createTestInvoiceContext(client, client.organization, dummyIntegration, { status: INVOICE_CONTEXT_STATUS_FINISHED })
+            const [createdInvoice] = await createTestInvoice(client, invoiceContext, {
+                property: { connect: { id: client.property.id } },
+                unitType,
+                unitName,
+                status: INVOICE_STATUS_DRAFT,
+                client: { connect: { id: residentClient.user.id } },
+            })
+
+            const invoice = await Invoice.update(adminClient, createdInvoice.id, {
+                status: INVOICE_STATUS_PUBLISHED,
+                dv: 1,
+                sender: { dv: 1, fingerprint: 'tests' },
+            })
+
+            const messages = await Message.getAll(adminClient, {
+                user: { id: residentClient.user.id },
+                type: MARKETPLACE_INVOICE_PUBLISHED_MESSAGE_TYPE,
+            })
+
+            expect(messages).toHaveLength(1)
+            expect(messages[0].meta.data.invoiceId).toEqual(invoice.id)
+            expect(messages[0].meta.data.residentId).toEqual(resident.id)
+            expect(messages[0].meta.data.url).toEqual(`${conf.SERVER_URL}/invoice/${invoice.id}`)
+        })
+
+        test('not send push after create invoice with no published status', async () => {
+            const client = await makeClientWithProperty()
+
+            const unitType = FLAT_UNIT_TYPE
+            const unitName = faker.lorem.word()
+
+            const residentClient = await makeClientWithResidentUser()
+            const [resident] = await registerResidentByTestClient(
+                residentClient,
+                {
+                    address: client.property.address,
+                    addressMeta: client.property.addressMeta,
+                    unitType,
+                    unitName,
+                })
+
+            const [invoiceContext] = await createTestInvoiceContext(client, client.organization, dummyIntegration, { status: INVOICE_CONTEXT_STATUS_FINISHED })
+            await createTestInvoice(client, invoiceContext, {
+                property: { connect: { id: client.property.id } },
+                unitType,
+                unitName,
+                status: INVOICE_STATUS_PAID,
+                client: { connect: { id: residentClient.user.id } },
+            })
+
+            const messages = await Message.getAll(adminClient, {
+                user: { id: residentClient.user.id },
+                type: MARKETPLACE_INVOICE_PUBLISHED_MESSAGE_TYPE,
+            })
+
+            expect(messages).toHaveLength(0)
```

Comment:
At least all of the `Message` domain related logic should be tested with delayed helper function `waitFor`